### PR TITLE
[risk=low][no ticket] Followups for access tier expiration emails

### PR DESCRIPTION
**Pull request by jmthibault79**
Followups to #7429 

Not sure if all of these are desirable.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/access/AccessSyncServiceImpl.java [hunk 1]
```diff
@@ -101,7 +104,7 @@ private List<DbAccessTier> getUserAccessTiersList(DbUser dbUser) {
   }
 
   private boolean shouldGrantUserTierAccess(
-      DbUser user, List<DbAccessModuleName> requiredModules, String tierShortName) {
+      DbUser user, Collection<DbAccessModuleName> requiredModules, String tierShortName) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessSyncServiceImpl.java:107]: we're not using specific `List` features, so the more-generic type is adequate

File: api/src/main/java/org/pmiops/workbench/access/AccessUtils.java [hunk 2]
```diff
@@ -34,13 +37,25 @@ public static DbAccessTier getAccessTierByShortNameOrThrow(
...
-  public static final List<DbAccessModuleName> REQUIRED_MODULES_FOR_CONTROLLED_TIER =
-      ImmutableList.of(
-          DbAccessModuleName.CT_COMPLIANCE_TRAINING,
-          DbAccessModuleName.DATA_USER_CODE_OF_CONDUCT,
-          DbAccessModuleName.PROFILE_CONFIRMATION,
-          DbAccessModuleName.PUBLICATION_CONFIRMATION,
-          DbAccessModuleName.RAS_LOGIN_GOV,
-          DbAccessModuleName.RT_COMPLIANCE_TRAINING,
-          DbAccessModuleName.TWO_FACTOR_AUTH);
+  public static Collection<DbAccessModuleName> getRequiredModulesForRegisteredTierAccess() {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessUtils.java:40]: these 2 are intentionally identical.  Does that deserve a comment?

File: api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java [hunk 3]
```diff
@@ -823,20 +823,20 @@ public void signOut(DbUser user) {
   }
 
   /**
-   * Return the user's registered tier access expiration time, for the purpose of sending an access
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java:826]: not just RT anymore

File: api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java [hunk 4]
```diff
@@ -823,20 +823,20 @@ public void signOut(DbUser user) {
...
-   * renewal reminder or expiration email.
+   * Return the user's access expiration time for a given set of modules, for the purpose of sending
+   * an access renewal reminder or expiration email.
    *
    * <p>First: ignore any bypassed modules. These are in compliance and do not need to be renewed.
    *
    * <p>Next: do all un-bypassed modules have expiration times? If yes, return the min (earliest).
-   * If no, either the AAR feature flag is not set or the user does not have access for reasons
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java:832]: feature flag no longer exists

File: api/src/main/resources/emails/README.md [hunk 5]
```diff
@@ -27,7 +27,7 @@ There are two commands that you should be aware of.
 
 First, in order to convert an mjml file to an html file, you can run:
 
-```mjml index.mjml index.html```
+```mjml index.mjml > index.html```
```
Comment (jmthibault79) [inline api/src/main/resources/emails/README.md:30]: Not sure if we should change it to this, but it was required for me.

File: api/src/main/resources/emails/tier_access_expired/content.html [hunk 6]
```diff
@@ -1,3 +1,4 @@
+<!-- FILE: content.mjml -->
```
Comment (jmthibault79) [inline api/src/main/resources/emails/tier_access_expired/content.html:1]: My version of mjml added this

File: api/src/main/resources/emails/tier_access_threshold/content.html [hunk 7]
```diff
@@ -163,7 +164,7 @@
                   <tbody>
                     <tr>
                       <td align="left" style="font-size:0px;padding:10px 25px;word-break:break-word;">
-                        <div style="font-family:'Montserrat', sans-serif;font-size:24;font-weight:600;line-height:1.5;text-align:left;color:#000000;">Hello ${FIRST_NAME},</div>
+                        <div style="font-family:'Montserrat', sans-serif;font-size:16px;font-weight:600;line-height:1.5;text-align:left;color:#000000;">Hello ${FIRST_NAME},</div>
```
Comment (jmthibault79) [inline api/src/main/resources/emails/tier_access_threshold/content.html:167]: my version of mjml said "24" was invalid as a size; px was required.

I updated it to 16px to match the other email.

File: api/src/main/resources/emails/tier_access_expired/content.html (hunk 6 above)
Comment (evrii) [reply api/src/main/resources/emails/tier_access_expired/content.html:1]: That's strange. Did you install mjml globally? if so, which version does it say you have in ```vim /usr/local/lib/node_modules/mjml/package.json```

**Review by evrii — APPROVED** [review]
Looks good to me. Thanks for making these changes.

File: api/src/main/java/org/pmiops/workbench/access/AccessUtils.java (hunk 2 above)
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/access/AccessUtils.java:40]: We need a function for the CT version (at least the way I have done it here) so I made RT also functions, for consistency

File: api/src/main/resources/emails/README.md (hunk 5 above)
Comment (jmthibault79) [reply api/src/main/resources/emails/README.md:30]: -o is better.  Updated.

File: api/src/main/resources/emails/tier_access_expired/content.html (hunk 6 above)
Comment (jmthibault79) [reply api/src/main/resources/emails/tier_access_expired/content.html]: -o removes it, so now my version matches.  Updated.

**Pull request merged**